pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -488,20 +467,35 @@ func (nc *Controller) doFixDeprecatedTaintKeyPass(node *v1.Node) error {
 	return nil
 }
 
-func (nc *Controller) doNoScheduleTaintingPassWorker(i int, stopCh <-chan struct{}) {
+func (nc *Controller) doNoScheduleTaintingPassWorker() {
 	for {
-		select {
-		case <-stopCh:
+		obj, shutdown := nc.nodeUpdateQueue.Get()
+		// "nodeUpdateQueue" will be shutdown when "stopCh" closed;
+		// we do not need to re-check "stopCh" again.
+		if shutdown {
 			return
-		case node := <-nc.nodeUpdateChannels[i]:
-			if err := nc.doNoScheduleTaintingPass(node); err != nil {
-				glog.Errorf("Failed to taint NoSchedule on node <%s>: %v", node.Name, err)
-			}
 		}
+		nodeName := obj.(string)
+
+		if err := nc.doNoScheduleTaintingPass(nodeName); err != nil {
+			glog.Errorf("Failed to taint NoSchedule on node <%s>: %v", nodeName, err)
```

Comment:
If at some point we decide to not update node status that frequently, this may become a problem (e.g. if you fail to send patch request).

Please add a todo to requeue this node name again in that case.